tokio-postgres/src/config.rs

```diff
@@ -347,14 +359,53 @@ impl Config {
     ///
     /// This is ignored for Unix domain sockets, or if the `keepalives` option is disabled. Defaults to 2 hours.
     pub fn keepalives_idle(&mut self, keepalives_idle: Duration) -> &mut Config {
-        self.keepalives_idle = keepalives_idle;
+        self.keepalive_config.idle = keepalives_idle;
         self
     }
 
     /// Gets the configured amount of idle time before a keepalive packet will
     /// be sent on the connection.
     pub fn get_keepalives_idle(&self) -> Duration {
-        self.keepalives_idle
+        self.keepalive_config.idle
+    }
+
+    /// Sets the time interval between TCP keepalive probes.
+    /// On Windows, this sets the value of the tcp_keepalive struct’s keepaliveinterval field.
+    ///
+    /// This is ignored for Unix domain sockets, or if the `keepalives` option is disabled.
+    ///
+    /// Available on neither Redox nor Solaris.
+    #[cfg(not(any(target_os = "redox", target_os = "solaris")))]
```

Comment:
I think it'd be preferable to always define these fields and just not do anything with them when unsupported. Otherwise the crate's public API is tied to the platform support in socket2.

Reply:
I fixed it.